modules/recon/masscan/masscan.go

```diff
@@ -0,0 +1,347 @@
+	log.Debug("H3ll-0 M4sscan")
+
+	// Temporary IP forced : 212.47.247.190 = edznux.fr
+	outputfile := "/tmp/" + generateUUID() + ".json"
+
+	// Get arguments
+	log.Debug("Get arguments")
+	opt, err := M.ParseOptions()
+	if err != nil {
+		log.Fatal(err)
+	}
+	opt = append(opt, "-oJ", outputfile)
+
+	// Get IP
+	log.Debug("Get IP")
+	for _, i := range inputs {
+		if i.IP != nil {
+			opt = append([]string{i.IP.String()}, opt...)
+		}
+	}
+
+	// Command execution
+	cmd := exec.Command("masscan", opt...)
+	log.Debug("Command executed: masscan ", opt)
+	var stdout bytes.Buffer
+	cmd.Stdout = &stdout
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+
+	err = cmd.Run()
+	if err != nil {
+		log.Info(stderr.String())
+	}
+	log.Debug(stdout.String())
+	res, err := M.Parse(outputfile)
+	log.Debug("res:%+v", res)
+	log.Debug("err:%+v", nil)
+	if err != nil {
+		log.Error(err)
+	}
+	log.Debug("M4sscan done.")
+
+	return modules.Result{Data: res, Timestamp: time.Now(), Module: M.Name()}, nil
+}
+
+// ParseConfig : Load the config from the config folder
+func (M *Masscan) ParseConfig() error {
+	ex, err := os.Executable()
+	if err != nil {
+		log.Panic(err)
+	}
+	exPath := filepath.Dir(ex)
+	configPath := filepath.Join(exPath, "config", "masscan.conf")
+
+	if _, err := toml.DecodeFile(configPath, &M.Config); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Parse : Parse the result of the execution
+func (M *Masscan) Parse(file string) (MasscanResult, error) {
+	var scans []Scan
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		return MasscanResult{}, err
+	}
+	fileReformatted := string(data)
+
+	// JSON reformatted
+	re := regexp.MustCompile(" },\n]\n")
+	fileReformatted = re.ReplaceAllString(string(data), " }\n]\n")
+
+	err = json.Unmarshal([]byte(fileReformatted), &scans)
+	if err != nil {
+		return MasscanResult{}, err
+	}
+
+	err = os.Remove(file)
+	if err != nil {
+		log.Error(err)
+	}
+	return MasscanResult{Resultat: scans}, nil
+}
+
+// WriteDb : Save data
+func (M *Masscan) WriteDb(result modules.Result, db models.Database, projectName string) error {
+	log.Info("Write to the database.")
+
+	var data MasscanResult
+	data = result.Data.(MasscanResult)
+
+	var ips []models.IP
+
+	for _, itemIP := range data.Resultat {
+		ipn := models.IP{Value: itemIP.IP}
+		ips = append(ips, ipn)
+		for _, itemPort := range itemIP.Ports {
+			portn := models.Port{
+				Number:   int16(itemPort.Port),
+				Protocol: itemPort.Proto,
+				Status:   itemPort.Status,
+				Banner:   itemPort.Service.Banner}
+			err := db.CreateOrUpdatePort(M.Name(), itemIP.IP, portn)
+			if err != nil {
+				log.Errorf("Could not create or update port : %+v", err)
+			}
+		}
+	}
+
+	err := db.CreateOrUpdateIPs(M.Name(), ips)
+	if err != nil {
+		log.Errorf("Could not create or update ip : %+v", err)
+	}
+
+	err = db.AppendRawData(projectName, M.Name(), data)
+	if err != nil {
+		log.Errorf("Could not append : %+v", err)
+	}
+
+	return nil
```

Comment:
You only return nil there
You should return err in every `if err != nil`. Maybe with errors.New("message ..."+ err.Error()) instead of / in addition to log.Errorf